cgmes/cgmes-extensions/src/main/java/com/powsybl/cgmes/extensions/CgmesIidmMappingImpl.java

```diff
@@ -115,7 +123,46 @@ public CgmesIidmMapping put(String busId, String topologicalNodeId) {
         if (busTopologicalNodeMap.isEmpty()) {
             calculate();
         }
-        return new HashSet<>(unmapped);
+        return new HashSet<>(unmappedTopologicalNodes);
+    }
+
+    @Override
+    public Map<Double, BaseVoltageSource> getBaseVoltages() {
+        return Collections.unmodifiableMap(nominalVoltageBaseVoltageMap);
+    }
+
+    @Override
+    public BaseVoltageSource getBaseVoltage(double nominalVoltage) {
+        return nominalVoltageBaseVoltageMap.get(nominalVoltage);
+    }
+
+    @Override
+    public boolean isBaseVoltageMapped(double nominalVoltage) {
+        return nominalVoltageBaseVoltageMap.containsKey(nominalVoltage);
+    }
+
+    @Override
+    public boolean isBaseVoltageEmpty() {
+        return nominalVoltageBaseVoltageMap.isEmpty();
+    }
+
+    @Override
+    public CgmesIidmMapping addBaseVoltage(double nominalVoltage, String baseVoltageId, Source source) {
+        if (nominalVoltageBaseVoltageMap.containsKey(nominalVoltage)) {
+            if (nominalVoltageBaseVoltageMap.get(nominalVoltage).getSource().equals(Source.IGM) && source.equals(Source.BOUNDARY)) {
+                LOGGER.info("Nominal voltage {} is already mapped with an {} base voltage. Replaced by a {} base voltage", nominalVoltage, Source.IGM.name(), Source.BOUNDARY.name());
+                nominalVoltageBaseVoltageMap.put(nominalVoltage, new BaseVoltageSource(baseVoltageId, nominalVoltage, source));
+            } else {
+                LOGGER.info("Nominal voltage {} is already mapped and not to the given base voltage {} from {}", nominalVoltage, baseVoltageId, source.name());
+            }
```

Comment:
Before entering here, check if Logger level is INFO

Reply:
At this point we try to make sure that the mapped BaseVoltage belongs to the boundaries file, replacing the baseVoltage mapped if necessary. This is not just log information

Reply:
Ah yes, you're right! Sorry, my bad.